chain/store/name.go

```diff
@@ -168,39 +168,50 @@ func (sdb *StateDB) deleteNameInfo(nameId string) error {
 	return nil
 }
 
-func (sdb *StateDB) FinalizeNames(commit bool) {
+func (sdb *StateDB) FinalizeNames(commit bool) error {
 	_, height, _ := sdb.cs.getCurrentBlockHashFromDB()
 	if config.LegacyNameService.GetValueAtHeight(height + 1) {
-		sdb.FinalizeNames_legacy(commit)
-		return
+		return sdb.FinalizeNames_legacy(commit)
 	}
+
+	var err error
 	sdb.names.Range(func(key, value interface{}) bool {
 		if nameId, ok := key.(string); ok {
 			if info, ok := value.(*nameInfo); ok && !info.Empty() {
-				sdb.updateNameInfo(nameId, info)
+				err = sdb.updateNameInfo(nameId, info)
 			} else {
-				sdb.deleteNameInfo(nameId)
+				err = sdb.deleteNameInfo(nameId)
+			}
+			if err != nil {
+				return false
 			}
 			if commit {
 				sdb.names.Delete(nameId)
 			}
 		}
 		return true
 	})
+	if err != nil {
+		return err
+	}
 
 	sdb.namesCleanup.Range(func(key, value interface{}) bool {
 		if height, ok := key.(uint32); ok {
 			if nc, ok := value.(namesCleanup); ok && len(nc) > 0 {
-				sdb.updateNamesCleanup(height, nc)
+				err = sdb.updateNamesCleanup(height, nc)
 			} else {
-				sdb.deleteNamesCleanup(height)
+				err = sdb.deleteNamesCleanup(height)
+			}
+			if err != nil {
+				return false
 			}
 			if commit {
 				sdb.namesCleanup.Delete(height)
 			}
 		}
 		return true
 	})
+	return err
```

Comment:
Maybe `return nil` here is more clear, as err can only be `nil`.

Reply:
`err` can be non nil. Return at L206 is not for `FinalizeNames` but for `sdb.namesCleanup.Range` iterator.